velox/functions/lib/CMakeLists.txt

```diff
@@ -24,7 +24,8 @@ velox_link_libraries(velox_functions_util velox_vector velox_common_base)
 velox_add_library(velox_functions_lib_date_time_formatter DateTimeFormatter.cpp
                   DateTimeFormatterBuilder.cpp)
 
-velox_link_libraries(velox_functions_lib_date_time_formatter velox_type_tz)
+velox_link_libraries(velox_functions_lib_date_time_formatter velox_type_tz
+                     ICU::i18n ICU::uc)
```

Comment:
ICU is only available as a target if we build boost from source. As functions/lib is always used please move these lines https://github.com/facebookincubator/velox/blob/main/CMake/resolve_dependency_modules/boost.cmake#L25-L34
Into the root cmake file to have ICU always available.